### ISSUE TITLE
`musl` build

### ISSUE THREAD

**Issue by notramo** [Exafunction/codeium#3]
Please provide `musl` based build for Linux, or compile a fully static build.

**Comment by pqn** [comment]
Curious, what's your setup that would make a `musl` build helpful?

**Comment by notramo** [comment]
I'm using Void Linux `musl`, but there are several `musl` based Linux distros, probably the most popular are Alpine and postmarketOS.

**Comment by notramo** [comment]
@pqn what are the plans on this? Should I spend time on setting up another code recommendation system, or can I expect a `musl` version to be released in the near future?

**Comment by pqn** [comment]
I don't think we can offer one soon. We have quite a few dependencies and would have to work through the musl compatibility for them.

**Comment by DieracDelta** [comment]
I would also be interested in a musl build. IMO this should be doable with `nix` (or at least, I've done this for a bunch of software previously with nix). Do you have insights on which deps would be difficult? I'd be interested in helping.

As for my usecase: being able to use codeium on any x86/aarch64 linux box without needing to fumble around with glibc versions.

**Comment by pqn** [comment]
Notably, we have a single Bazel C++ toolchain with CUDA support, and I don't think CUDA plays well with musl. But there is obviously no CUDA usage client-side, so it's not impossible. Thanks for offering to help -- I think most of the work is build system related and difficult to carve out, but if I notice any dependencies without support that we can't migrate away from, I'll point them out in this thread if anyone would like to investigate.

**Comment by wangchao-sh** [comment]
@pqn Please help provide a statically compiled version so that it can be used properly on systems with 'musl' toolchains like Alpine.

**Comment by wangchao-sh** [comment]
@notramo May I know how you ultimately solved it? I also hope to be able to use codeium on Alpine.

**Comment by notramo** [comment]
Try this: https://github.com/copilot

**Comment by DieracDelta** [comment]
@wangchao-sh I think you can still run this on alpine using dynamic linking to stuff in the nix store using a statically linked nix binary. Patchelf -> `nix run` botta bing botta boom.

**Comment by PillFall** [comment]
@pqn Using Codeium in musl-based Linux distros also improves DX on [DevContainers](https://containers.dev/) heavily based on AlpineLinux, as the extension installs on the "Remote Side" (a.k.a. inside the container)

**Comment by notramo** [comment]
This LSP plugin can use Codeium: https://github.com/leona/helix-gpt
